CMakeLists.txt

```diff
@@ -0,0 +1,198 @@
+
+# Get the git version
+version_from_git(LOG ON TIMESTAMP "%Y%m%d%H%M%S")
+
+project(
+  PcapPlusPlus
+  DESCRIPTION
+    "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets."
+  VERSION ${VERSION})
+set(CMAKE_PROJECT_HOMEPAGE_URL "https://pcapplusplus.github.io/")
+
+# Include the other modules
+include(CMakePackageConfigHelpers)
+include(DetectCompiler)
+include(DetectOS)
+include(GNUInstallDirs)
+set(PCAPP_INSTALL_LIBDIR ${CMAKE_INSTALL_LIBDIR}/pcapplusplus)
+set(PCAPP_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR}/pcapplusplus)
+set(PCAPP_INSTALL_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/pcapplusplus)
+
+# Build options (Turn on Examples and Tests if it's the main project)
+option(PCAPPP_BUILD_EXAMPLES "Build Examples" ${PCAPPP_MAIN_PROJECT})
+option(BUILD_TESTING "Build Tests" ${PCAPPP_MAIN_PROJECT})
+option(BUILD_SHARED_LIBS "Build using shared libraries" OFF)
+
+option(PCAPPP_USE_DPDK
+       "Setup PcapPlusPlus with DPDK. In this case you must also set DPDK_HOME")
+option(
+  PCAPPP_USE_PF_RING
+  "Setup PcapPlusPlus with PF_RING. In this case you must also set PF_RING_HOME"
+)
+option(PCAPPP_INSTALL "Install Pcap++" ${PCAPPP_MAIN_PROJECT})
+
+# Set C++11
+set(CMAKE_CXX_STANDARD 11)
+# popen()/pclose() are not C++ standards
+set(CMAKE_CXX_EXTENSIONS ON)
+
+# Do we need Packet Packages on Windows ?
+if(TARGET_PLATFORM_WINDOWS)
+  find_package(Packet)
+  if(NOT Packet_FOUND)
+    message(FATAL_ERROR "Packet library -DPacket_ROOT=<NPCAP_SDK_PATH>")
+  endif()
+endif()
+
+find_package(PCAP)
+if(NOT PCAP_FOUND)
+  if(TARGET_PLATFORM_WINDOWS)
+    message("Specify NPCAP Library with -DPCAP_ROOT=<NPCAP_SDK_PATH>")
+  else()
+    message(FATAL_ERROR "PCAP library not found!")
+  endif()
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_IMMEDIATE_MODE
+       "Enable Pcap Has Immediate Mode (supported on libpcap>=1.5)"
+       ${HAVE_PCAP_IMMEDIATE_MODE})
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_IMMEDIATE_MODE)
+  if(NOT HAVE_PCAP_IMMEDIATE_MODE)
+    message(FATAL_ERROR "PCAP library doesn't have Immediate Mode support!")
+  endif()
+  add_definitions(-DHAS_PCAP_IMMEDIATE_MODE)
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_SET_DIRECTION "Enable Pcap Direction catpure"
+       ${HAVE_PCAP_DIRECTION})
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_SET_DIRECTION)
+  if(NOT HAVE_PCAP_DIRECTION)
+    message(FATAL_ERROR "PCAP library doesn't have Direction support!")
+  endif()
+  #add_definitions(-DHAS_SET_DIRECTION_ENABLED)
+endif()
+
+if(TARGET_PLATFORM_WINDOWS AND TARGET_COMPILER_MSVC)
+  # Other hacks for VStudio
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+else()
+  set(THREADS_PREFER_PTHREAD_FLAG ON)
+  find_package(Threads REQUIRED)
+  if(NOT CMAKE_USE_PTHREADS_INIT)
+    message(FATAL_ERROR "Pthreads not found!")
+  endif()
+  set(_PCAPP_THREADS Threads::Threads)
```

Comment:
I'm not sure all of this is needed in C++11 because threads are part of the standard library...

If we do need it, I'm not sure why we need to define the `_PCAPP_THREADS` variable... can't we just use `Threads::Threads`? 🤔

Reply:
That's what a old variable introduced to support pthread-win32